dbms/programs/client/Client.cpp

```diff
@@ -722,7 +722,11 @@ class Client : public Poco::Util::Application
 
                 try
                 {
-                    if (!processSingleQuery(str, ast) && !ignore_error)
+                    auto ast_to_process = ast;
+                    if (insert && insert->data)
+                        ast_to_process = nullptr;
```

Comment:
Does it mean, that the client will send data in source format uncompressed?

Reply:
This code contained an error. If client were used in `multiquery` mode, than all queries after insert 
``into tbl values (1, 2, 3);
select * from system.numbers limit 10;
...etc...
```
were placed into `insert->data` field. But it didn't lead to problems, because server had ignored `data` filed of insert queries. Now server sometimes parse `insert->data`, so we have to process `multiquery` insert queries correctly on client side.  
I don't know how this is connected with compression.

Reply:
Queries were always sent to server without embedded data.

Reply:
When data is sent in blocks (not embedded in query), it is compressed.

Reply:
`Queries were always sent to server without embedded data.`
This is true just for `insert into ... values` queries of `clickhouse-client` https://github.com/yandex/ClickHouse/blob/master/dbms/programs/client/Client.cpp#L883-L885. If there were `insert-select query`, more precisely buggy `insert .. values (1, 2, 3); select * from ...` query parsed with `multiquery` option, it was sent with embedded `query->data`, but `data` was always ignored on server side: https://github.com/yandex/ClickHouse/blob/master./dbms/src/Interpreters/executeQuery.cpp#L168-L169. 

Since I add logic for embedded data processing on server side it became a problem, so I split parsing of insert queries (parse each query separately, not all of them together) if they are insert queries and contain embedded `data`.

Nothing have changed, native client (`clickhouse-client`) will send insert queries with not embedded data in compressed blocks. I still do not understand how specified line connected with compression of data. It just query parsing and there was bug for multiqueries.